pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,280 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
+        # has_index_names = is_list_like(header) and len(header) > 1
+
+        if skiprows:
+            data = [row for i, row in enumerate(data) if i not in skiprows]
+
+        if skipfooter:
+            data = data[:-skipfooter]
+
+        column_names = [cell for cell in data.pop(0)]
+
+        frame = DataFrame(data, columns=column_names)
+        frame = self._handle_usecols(frame, usecols)
+
+        if not converters:
+            converters = dict()
+        if not dtype:
+            dtype = dict()
+
+        # handle columns referenced by number so all references are by
+        # column name
+        handled_converters = {}
+        for k, v in converters.items():
+            if k not in frame.columns and isinstance(k, int):
+                k = frame.columns[k]
+            handled_converters[k] = v
+        converters = handled_converters
+
+        if len(frame) > 0:
+            for column in set(frame) - set(dtype.keys()):
+                frame[column] = self._handle_convert_float(frame[column],
+                                                           convert_float)
+
+        if converters:
+            for k, v in converters.items():
+                # for compatibiliy reasons
+                if frame[k].dtype == float and convert_float:
+                    frame[k] = frame[k].fillna('')
+                frame[k] = frame[k].apply(v)
+
+        if dtype:
+            for k, v in dtype.items():
+                frame[k] = frame[k].astype(v)
+
+        frame = self._handle_index_col(frame, index_col)
+
+        if not squeeze or isinstance(frame, DataFrame):
+            if header_names:
+                frame = frame.columns.set_names(header_names)
+
+        # name unnamed columns
+        unnamed = 0
+        for i, col_name in enumerate(frame.columns.values):
+            if col_name is None:
+                frame.columns.values[i] = "Unnamed: {n}".format(n=unnamed)
+                unnamed += 1
+        return frame
+
+    def parse(self,
+              sheet_name=0,
+              header=0,
+              names=None,
+              index_col=None,
+              usecols=None,
+              squeeze=False,
+              converters=None,
+              dtype=None,
+              true_values=None,
+              false_values=None,
+              skiprows=None,
+              nrows=None,
+              na_values=None,
+              verbose=False,
+              parse_dates=False,
+              date_parser=None,
+              thousands=None,
+              comment=None,
+              skipfooter=0,
+              convert_float=True,
+              mangle_dupe_cols=True,
+              **kwds):
+
+        _validate_header_arg(header)
+
+        sheets = self._handle_sheet_name(sheet_name)
+        ret_dict = len(sheets) != 1
+        output = OrderedDict()
+
+        for asheetname in sheets:
+            if asheetname in output.keys():
+                # skip duplicates in sheets
```

Comment:
Is this actually possible? Didn't think Excel allowed duplicate sheet names

Reply:
I don't know, but the `xlrd` code went out of its way to handle this case so I kept that.

Reply:
Yea I think OK to take out. I've never seen this in Excel anyway:

<img width="423" alt="image" src="https://user-images.githubusercontent.com/609873/57120372-db2eb680-6d25-11e9-9745-2194dd7d1b6c.png">